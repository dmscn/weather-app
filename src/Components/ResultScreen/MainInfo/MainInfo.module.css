.Container {
  display: flex;
  width: 70vw;
  justify-content: space-around;
  align-items: center;
  margin-top: 1em;
}

.Icon {
  font-size: 8em;
  font-weight: 200;
}

.weatherBox {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.WeatherTitle {
  font-size: 2em;
  font-weight: 200;
  margin: 0 0;
}

.Temperature {
  font-size: 2em;
  font-weight: 400;
  margin: .2em 0;
}

.MainTemperature {
  font-size: 3em;
  margin: .5em 0;
  font-weight: bold;
}
