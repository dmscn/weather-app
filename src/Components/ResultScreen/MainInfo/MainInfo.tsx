import React, { Component } from 'react'

import '../../../static/weather-icons/css/weather-icons.min.css';
import styles from './MainInfo.module.css';

export interface Props {
  weatherId: number,
  weather: string,
  temp: number,
  tempMin: number,
  tempMax: number,
}

export default class MainInfo extends Component<Props> {

  setIcon = (): JSX.Element => {
    const { weatherId } = this.props;
    const icon: string = `wi wi-owm-${weatherId}`;
    return <i className={icon} style={{fontSize: '8em'}}></i>
  }

  render(): JSX.Element {
    const { weather, temp, tempMin, tempMax } = this.props;
    return (
      <section className={styles.Container}>
        <h4>{weather}</h4>
        {this.setIcon()}
        <h4>{tempMin}</h4>
        <h4>{temp}</h4>
        <h4>{tempMax}</h4>
      </section>
    )
  }
}
