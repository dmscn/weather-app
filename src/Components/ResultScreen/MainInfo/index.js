export {default} from './MainInfo';
export * from './MainInfo';