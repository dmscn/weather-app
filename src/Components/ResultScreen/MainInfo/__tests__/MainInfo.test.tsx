import React from 'react'
import {shallow} from 'enzyme';

import MainInfo from '../MainInfo';

describe('<MainInfo />', () => {
  it('renders', () => {
    const wrapper = shallow(<MainInfo />);
    expect(wrapper).toMatchSnapshot();
  });
})