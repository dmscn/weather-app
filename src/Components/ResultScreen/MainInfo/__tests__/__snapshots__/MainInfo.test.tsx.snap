// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainInfo /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4 />,
        <i
          className="wi wi-owm-undefined"
          style={
            Object {
              "fontSize": "8em",
            }
          }
        />,
        <h4 />,
        <h4 />,
        <h4 />,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "wi wi-owm-undefined",
          "style": Object {
            "fontSize": "8em",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4 />,
          <i
            className="wi wi-owm-undefined"
            style={
              Object {
                "fontSize": "8em",
              }
            }
          />,
          <h4 />,
          <h4 />,
          <h4 />,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "wi wi-owm-undefined",
            "style": Object {
              "fontSize": "8em",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
